microsoft/testsuites/core/timesync.py

```diff
@@ -351,3 +352,33 @@ def timesync_chrony(self, node: Node) -> None:
         chrony.check_sources_and_stats()
         # 5. Check chrony tracking.
         chrony.check_tracking()
+
+    @TestCaseMetadata(
+        description="""
+        This test is to check gettime calls are not being redirected as system calls,
+        leading to performance bottleneck, Linux systems have a mechanism called vdso
+        which helps in gettimeoftheday and clock_gettime to be executed in userspace (no
+        syscall).
+
+        Steps:
+            1. Run strace date 2>&1
+            2. Check output of 1, should not show the time related syscalls,
+             e.g. clock_gettime, gettimeofday
+        """,
+        priority=2,
+    )
+    def verify_vdso(self, node: Node) -> None:
+        time_pattern = re.compile("(time|gettimeofday|clock_gettime)", re.M)
+        strace = node.tools[Strace]
+        # 1. Run strace date 2>&1
+        output = strace.get("date 2>&1")
```

Comment:
How about other vdso commands? We should cover as much as possible.

Reply:
- https://github.com/nlynch-mentor/vdsotest
- https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/tree/tools/testing/selftests/vDSO

I found 2 repos which can test vdso. After trying, found the second one can't test the bug. The first one is easy to compile and can repro the issue, also it contains other vdso commands, do you think it is feasible to use it?

Reply:
ok, we can use the first one. And add your explanation in suite description to explain why we don't use the kernel tests.

Reply:
update